docs/modelserving/v1beta1/transformer/collocation/README.md

```diff
@@ -0,0 +1,123 @@
+# Collocate transformer and predictor in same pod
+
+KServe by default deploys the Transformer and Predictor as separate services so, you can deploy them on different devices and scale independently.
+<br/>However, In below scenarios you can choose to collocate the transformer and predictor in the same pod.
+
+1. If your transformer is tightly coupled with the predictor and you want to do canary together.
+2. If you want to reduce sidecar resources.
+3. If you want to reduce networking latency.
+
+## Before you begin 
+
+1. Your ~/.kube/config should point to a cluster with [KServe installed](../../../../get_started/README.md#install-the-kserve-quickstart-environment).
+2. Your cluster's Istio Ingress gateway must be [network accessible](https://istio.io/latest/docs/tasks/traffic-management/ingress/ingress-control/).
+3. You can find the [code samples](https://github.com/kserve/kserve/tree/master/docs/samples/v1beta1/transformer/collocation) on kserve repository.
+
+## Deploy the InferenceService
+
+Since, the predictor and the transformer are in the same pod, they need to listen on different ports to avoid conflict. `Transformer` is configured to listen on port 8000 and 8081 
+while, `Predictor` listens on port 8080 and 8082. `Transformer` calls `Predictor` on port 8082 via local socket. 
+Deploy the `Inferenceservice` using the below command.
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
+metadata:
+  name: custom-transformer-collocation
+spec:
+  predictor:
+    containers:
+      - name: kserve-container
```

Comment:
We probably need to add a note for the limitation that currently you need to use the custom container spec for kserve model container, we can create an issue to support model spec like the normal case.

Reply:
done

Reply:
issue created https://github.com/kserve/kserve/issues/2909